### PR TITLE
Automated cherry pick of #3067: Fix issue where controller is watching a secret that is never

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #3067 on release-v1.33.

#3067: Fix issue where controller is watching a secret that is never

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Review by caseydavenport — APPROVED** [review]

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Pull request merged**
